sdk/python/kubeflow/metadata/metadata.py

```diff
@@ -14,294 +14,335 @@
 
 import datetime
 import json
-from kubeflow.metadata import openapi_client
-from kubeflow.metadata.openapi_client import Configuration, ApiClient, MetadataServiceApi
-
-"""
-This module contains the Python API for logging metadata of machine learning
+import ml_metadata
+from ml_metadata.metadata_store import metadata_store
+from ml_metadata.proto import metadata_store_pb2 as mlpb
+from typing import Any, List, Mapping, Optional, TypeVar, Union
+"""This module contains the Python API for logging metadata of machine learning
 workflows to the Kubeflow Metadata service.
 """
 
-WORKSPACE_PROPERTY_NAME = '__kf_workspace__'
-RUN_PROPERTY_NAME = '__kf_run__'
-ALL_META_PROPERTY_NAME = '__ALL_META__'
+# TypeArtifact is the type Model, Dataset, Metrics or any class that has
+# class variable ARTIFACT_TYPE_NAME of string and a serialization() method that
+# returns a metadata_store_pb2.Artifact.
+TypeArtifact = TypeVar('TypeArtifact', 'Model', 'DataSet', 'Metrics', Any)
+
+_WORKSPACE_PROPERTY_NAME = '__kf_workspace__'
+_RUN_PROPERTY_NAME = '__kf_run__'
+_ALL_META_PROPERTY_NAME = '__ALL_META__'
+
+
+class Store(object):
+  """Metadata Store that connects to the Metadata gRPC service."""
+
+  def __init__(self,
+               grpc_host: str = "metadata-grpc-service.kubeflow",
```

Comment:
Maybe the default value be the one that works for the default istio kubeflow deployment, such as metadata-grpc-service.kubeflow.svc.cluster.local? Not sure if my example host is exactly correct.

Reply:
`<service>.<namespace>` should work as DNS inside a k8s cluster. I've used this pattern in multiple notebooks running in KF.

Reply:
If it works, great. I saw the other file also uses the <service>.<namespace>(https://github.com/kubeflow/metadata/blob/master/watcher/dockerfiles/watcher.yaml#L21).

I've read k8s doc that the DNS resolves the service as my-svc.my-namespace.svc.cluster-domain.example(https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#a-records). Do you know if <service>.<namespace> is resolved by Istio DNS or something else? Happy to learn it.

Reply:
`<service>.<NS>` is a way how k8s resolves DNS. Please see
https://kubernetes.io/docs/concepts/services-networking/service/#dns

Reply:
Thanks